funcx_endpoint/funcx_endpoint/endpoint/rabbit_mq/task_queue_publisher.py

```diff
@@ -0,0 +1,77 @@
+import logging
+
+import pika
+
+logger = logging.getLogger(__name__)
+
+
+class TaskQueuePublisher:
+    """The TaskQueue is a direct rabbitMQ pipe that runs from the service
+    to the endpoint.
+    Multiple subscribers are disabled, with endpoints using exclusive consume
+    Publish side will set heartbeats
+    """
+
+    def __init__(
+        self,
+        endpoint_uuid: str,
+        pika_conn_params: pika.connection.Parameters,
+        exchange: str = "tasks",
+    ):
+        """
+        Parameters
+        ----------
+        endpoint_uuid: str
+             Endpoint UUID string used to identify the endpoint
+        pika_conn_params: pika.connection.Parameters
+             Pika connection parameters to connect to RabbitMQ
+        exchange: str
+             Exchange name. Default: "tasks"
+        """
+        self.endpoint_uuid = endpoint_uuid
+        self.queue_name = f"{self.endpoint_uuid}.tasks"
+        self.routing_key = f"{self.endpoint_uuid}.tasks"
+        self.params = pika_conn_params
+        self.params._heartbeat = 60
+        self.params._blocked_connection_timeout = 120
+
+        self.exchange = exchange
+        self.exchange_type = "direct"
+
+        self._conn = None
```

Comment:
```suggestion
        self._connection = None
```

In the other classes this is "connection", not "conn". Let's be consistent.

Reply:
This was marked resolved but I don't see any change here?